replication/backup_test.go

```diff
@@ -0,0 +1,50 @@
+package replication
+
+import (
+	"sync"
+	"os"
+	"time"
+	. "github.com/pingcap/check"
+	"github.com/siddontang/go-mysql/mysql"
+	"context"
+	"github.com/juju/errors"
+)
+
+func (t *testSyncerSuite) TestStartBackupEndInGivenTime(c *C) {
+	t.setupTest(c, mysql.MySQLFlavor)
+	
+	t.testExecute(c, "RESET MASTER")
+	
+	var wg sync.WaitGroup
+	wg.Add(1)
+	defer wg.Wait()
+	
+	go func() {
+		defer wg.Done()
+		
+		t.testSync(c, nil)
+		
+		t.testExecute(c, "FLUSH LOGS")
+		
+		t.testSync(c, nil)
+	}()
+	
+	os.RemoveAll("./var")
+	timeout := 2 * time.Second
+	
+	done := make(chan bool)
+	
+	go func() {
+		err := t.b.StartBackup("./var", mysql.Position{"", uint32(0)}, timeout)
+		c.Assert(err, IsNil)
+		done <- true
+	}()
+	failTimeout := 2 * timeout
```

Comment:
to make the test more stable, I think you should use a larger timeout here to ensure StartBackup timeout can work.